src/index.ts

```diff
@@ -1 +1,135 @@
-export const add = (a: number, b: number) => a + b;
+import { getGeneratorForPrimes } from './utils.js';
+import {
+  isProbablyPrime,
+  modAdd,
+  modInv,
+  modMultiply,
+  modPow,
+  prime,
+  randBetween,
+} from 'bigint-crypto-utils';
+
+export * from './utils.js';
+
+export type Ciphertext = [bigint, bigint];
+
+export class PublicKey {
+  primeP: bigint;
+  primeQ: bigint;
+  generator: bigint;
+  publicKey: bigint;
+
+  constructor(
+    primeP: bigint,
+    primeQ: bigint,
+    generator: bigint,
+    publicKey: bigint,
+  ) {
+    this.primeP = primeP;
+    this.primeQ = primeQ;
+    this.generator = generator;
+    this.publicKey = publicKey;
+  }
+
+  encrypt(
+    plaintext: bigint,
+    randomness: bigint = randBetween(
+      modAdd([this.primeP, -1n], this.primeP),
+      1n,
+    ),
+  ): [Ciphertext, bigint] {
+    if (plaintext === 0n) {
```

Comment:
We need this check because 0n * y^r mod p always equaly zero and therefore is not secure.
But: Let primeP be 5. Then m1 = 12 and m2 = 7 and m3 = -3 represent three different plaintexts. Still, they're all encrypted with the exact same ciphertext which might not be as secure. Therefore I'd suggest:
```suggestion
    if (plaintext <= 0n || plaintext >= this.primeP) {
```